dashboard/api.py

```diff
@@ -367,3 +380,127 @@ def format_courserun_for_dashboard(course_run, status_for_user, certificate=None
         # TODO: here goes the logic to pull the current grade  # pylint: disable=fixme
         pass
     return formatted_run
+
+
+def get_student_enrollments(user, edx_client):
+    """
+    Return cached enrollment data or fetch enrollment data first if necessary.
+    All CourseRun will have an entry for the user: this entry will contain Null
+    data if the user does not have an enrollment.
+
+    Args:
+        user (django.contrib.auth.models.User): A user
+        edx_client (EdxApi): EdX client to retrieve enrollments
+    Returns:
+        Enrollments: an Enrollments object from edx_api.
+            This may contain more enrollments than
+            what we know about in MicroMasters if more exist from edX,
+            or it may contain fewer enrollments if they don't exist for the course id in edX
+    """
+    # Data in database is refreshed after 5 minutes
+    now = datetime.datetime.now(tz=pytz.utc)
+    refresh_delta = now - datetime.timedelta(minutes=REFRESH_ENROLLMENT_CACHE_MINUTES)
+
+    with transaction.atomic():
+        course_ids = CourseRun.objects.filter(course__program__live=True).exclude(
+            Q(edx_course_key__isnull=True) | Q(edx_course_key__exact='')
+        ).values_list("edx_course_key", flat=True)
```

Comment:
this same query is used @ line 464. my first thought is that we should add a helper method on the CourseRun model (`get_live_course_ids` or something). if that isn't preferred, we should at least add a helper method in this module